core/batcher.go

```diff
@@ -0,0 +1,348 @@
+
+	"github.com/anoideaopen/foundation/core/cachestub"
+	"github.com/anoideaopen/foundation/core/telemetry"
+	"github.com/anoideaopen/foundation/core/types"
+	"github.com/anoideaopen/foundation/hlfcreator"
+	"github.com/anoideaopen/foundation/internal/config"
+	"github.com/anoideaopen/foundation/proto"
+	pb "github.com/golang/protobuf/proto" //nolint:staticcheck
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	BatcherBatchExecuteEvent                    = "batcherBatchExecute"
+	TxBatcherRequestType     BatcherRequestType = "tx"
+)
+
+type (
+	BatcherBatchExecuteRequestDTO struct {
+		Requests []BatcherRequest `json:"requests"`
+	}
+
+	BatcherRequestType string
+
+	BatcherRequest struct {
+		BatcherRequestID   string             `json:"batcher_request_id"`   // BatcherRequestID batcher request id
+		Channel            string             `json:"channel"`              // Channel on which the chaincode will be invoked
+		Chaincode          string             `json:"chaincode"`            // Chaincode of the chaincode to invoke
+		Method             string             `json:"function"`             // Method of the chaincode function to invoke
+		Args               []string           `json:"args"`                 // Args to pass to the chaincode function
+		BatcherRequestType BatcherRequestType `json:"batcher_request_type"` // BatcherRequestType for choose method to handling request. for example tx, swaps, swaps_keys, multi_swaps, multi_swaps_keys
+	}
+
+	Batcher struct {
+		batchCacheStub *cachestub.BatchCacheStub
+		cc             *ChainCode
+		cfgBytes       []byte
+		ski            string
+		tracingHandler *telemetry.TracingHandler
+	}
+)
+
+// BatcherHandler allow execute few subtransaction (batch request) in single transaction in hlf using cache state between requests to solbe mcc problem
+// The arguments of this method contains array of requests for execution each request contain own arguments for call method
+func BatcherHandler(
+	traceCtx telemetry.TraceContext,
+	stub shim.ChaincodeStubInterface,
+	cfgBytes []byte,
+	creatorSKI, hashedCert [32]byte,
+	arguments []string,
+	cc *ChainCode,
+) ([]byte, error) {
+	tracingHandler := cc.contract.TracingHandler()
+	traceCtx, span := tracingHandler.StartNewSpan(traceCtx, BatcherBatchExecute)
+	defer span.End()
+
+	logger := Logger()
+	batchID := stub.GetTxID()
+	span.SetAttributes(attribute.String("batch_tx_id", batchID))
+	start := time.Now()
+	defer func() {
+		logger.Infof("batch %s elapsed time %d ms", batchID, time.Since(start).Milliseconds())
+	}()
+
+	if len(arguments) != 1 {
+		return nil, fmt.Errorf("expected 1 argument, got %d", len(arguments))
+	}
+
+	var batchDTO BatcherBatchExecuteRequestDTO
+	if err := json.Unmarshal([]byte(arguments[0]), &batchDTO); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal BatcherBatchExecuteRequestDTO: %w", err)
+	}
```

Comment:
var batchRequest BatchExecuteRequest

https://google.github.io/styleguide/go/decisions#variable-names

Reply:
@muzykantov I did it, thank you